x-pack/filebeat/input/s3/input.go

```diff
@@ -372,15 +377,42 @@ func (p *s3Input) handleS3Objects(svc s3iface.ClientAPI, s3Infos []s3Info, errC
 		objectHash := s3ObjectHash(s3Info)
 
 		// read from s3 object
-		reader, err := p.newS3BucketReader(svc, s3Info)
+		resp, err := p.getS3ObjectResponse(svc, s3Info)
 		if err != nil {
 			err = errors.Wrap(err, "newS3BucketReader failed")
+			p.logger.Error(err)
 			s3Context.setError(err)
 			return err
 		}
 
-		if reader == nil {
-			continue
+		reader := bufio.NewReader(resp.Body)
```

Comment:
you can extract this whole block to a new function so you can use `defer`, it will also make the code more readable

Reply:
Yes, I moved this piece of code into a separate function called createEventsFromS3Info